pyrep/robots/arms/arm.py

```diff
@@ -37,6 +38,56 @@ def __init__(self, count: int, name: str, num_joints: int,
         self._collision_collection = vrep.simGetCollectionHandle(
             '%s_arm%s' % (name, suffix))
 
+    def get_configs_for_tip_pose(self, position: List[float],
+                                 euler: List[float] = None,
+                                 quaternion: List[float] = None,
+                                 ignore_collisions=False,
+                                 trials=300, max_configs=60):
+
+        """Gets a valid joint configuration for a desired end effector pose.
+
+        Must specify either rotation in euler or quaternions, but not both!
+
+        :param position: The x, y, z position of the target.
+        :param euler: The x, y, z orientation of the target (in radians).
+        :param quaternion: A list containing the quaternion (x,y,z,w).
+        :param ignore_collisions: If collision checking should be disabled.
+        :param trials: The maximum number of attempts to reach max_configs
+        :param max_configs: The maximum number of configurations we want to
+            generate before ranking them.
+        :raises: ConfigurationError if no joint configuration could be found.
+
+        :return: A list of valid joint configurations for the desired end effector pose.
```

Comment:
Perhaps mention that they are sorted based on configuration distance?

Reply:
I don't think they are sorted based on configuration distance. The function [findSeveralCollisionFreeConfigsAndCheckApproach ](https://github.com/stepjam/PyRep/blob/master/pyrep/backend/vrepAddOnScript_PyRep.lua#L74)just continually appends the found configurations, regardless of their distance in configuration space.

Although the function on this [line](https://github.com/stepjam/PyRep/blob/master/pyrep/backend/vrepAddOnScript_PyRep.lua#L98), which returns a single configuration, does so based on minizing a metric in configuration space.

I can make some small modifications to findSeveralCollisionFreeConfigsAndCheckApproach to return a sorted list based on configuration distance though if this would be best?

Reply:
Yes I think it makes sense for it to be sorted. Maybe the best option would be to have a parameter 'sorted' (True by default), and then sort based on that?

Reply:
I've made the lua function sort the configs by default, and added testing for this (y)

Reply:
Great. Thanks!